File: The Diwali Shopping/TDS.cpp
#include <bits/stdc++.h>
using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */  
    int n;
    cin >> n;
    int arr[n];
    for( int i=0; i<n; i++){
        cin >> arr[i];
    }
    int offerMoney;
    cin >> offerMoney;
    sort(arr, arr+n);
    int totalAmount = 0;
    for( int i=0; i<n; i++){
        if( totalAmount < offerMoney ){
            totalAmount += arr[i];
        }else{
            break;
        }
    }
    if( totalAmount < offerMoney ){
        cout << "You cannot avail offer";
    }else{
        float payableAmout = totalAmount - (totalAmount * 30)/100.0;
        cout << fixed << setprecision(2) << payableAmout;
    }
    return 0;
}

File: The Diwali Shopping/TDS.java
package CodeFiesta;

import java.util.*;

public class TDS {

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        Scanner sc = new Scanner(System.in);
        int n=sc.nextInt();
        int arr[]=new int[n];
        for(int i=0;i<n;i++){
            arr[i]=sc.nextInt();
        }
        int p=sc.nextInt();
        int sum=0;
        for(int i=0;i<n;i++){
            sum=sum+arr[i];
        }
        Arrays.sort(arr);
        double ans=0;
        int sumAns=0;
        double max1=1000000;
        if(sum<p){
            System.out.println("You cannot avail offer");
        }
        else{
            for(int i=0;i<n;i++){
                sumAns=sumAns+arr[i];
                if(sumAns>=p){
                    ans=(double)sumAns*70/100;
                    // System.out.printf("%.2f",ans);
                    if(ans<=max1){
                        max1=(float)ans; 
                    }
                }
            }
            
            System.out.printf("%.2f",max1);
        }
        sc.close();
    }
}